internal/operands/tekton-tasks/tekton-tasks_test.go

```diff
@@ -0,0 +1,412 @@
+	"github.com/kubevirt/tekton-tasks-operator/pkg/common"
+	tektonbundle "github.com/kubevirt/tekton-tasks-operator/pkg/tekton-bundle"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	pipeline "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	namespace = "kubevirt"
+	name      = "test-tekton"
+)
+
+var _ = Describe("environments", func() {
+	var tt *tektonTasks
+	var mockedRequest *common.Request
+	BeforeEach(func() {
+		tt = getMockedTektonTasksOperand()
+		mockedRequest = getMockedRequest()
+	})
+
+	It("New function should return object with correct tasks", func() {
+		res := New(getMockedTestBundle())
+		Expect(len(res.tasks)).To(Equal(11), "should return correct number of tasks")
+		Expect(len(res.serviceAccounts)).To(Equal(11), "should return correct number of service accounts")
+		Expect(len(res.roleBindings)).To(Equal(11), "should return correct number of role bindings")
+		Expect(len(res.clusterRoles)).To(Equal(11), "should return correct number of cluster roles")
+		for _, task := range res.tasks {
+			if _, ok := AllowedTasks[task.Name]; !ok {
+				Expect(ok).To(BeTrue(), "only allowed task is deployed - "+task.Name)
+			}
+		}
+		for _, sa := range res.serviceAccounts {
+			if _, ok := AllowedTasks[strings.TrimSuffix(sa.Name, "-task")]; !ok {
+				Expect(ok).To(BeTrue(), "only allowed service accounts is deployed - "+sa.Name)
+			}
+		}
+		for _, rb := range res.roleBindings {
+			if _, ok := AllowedTasks[strings.TrimSuffix(rb.Name, "-task")]; !ok {
+				Expect(ok).To(BeTrue(), "only allowed role bindings is deployed - "+rb.Name)
+			}
+		}
+		for _, cr := range res.clusterRoles {
+			if _, ok := AllowedTasks[strings.TrimSuffix(cr.Name, "-task")]; !ok {
+				Expect(ok).To(BeTrue(), "only allowed role bindings is deployed - "+cr.Name)
+			}
+		}
+	})
+
+	It("Name function should return correct name", func() {
+		name := tt.Name()
+		Expect(name).To(Equal(operandName), "should return correct name")
+	})
+
+	It("Reconcile function should return correct functions", func() {
+		functions, err := tt.Reconcile(mockedRequest)
+		Expect(err).ToNot(HaveOccurred(), "should not throw err")
+		Expect(len(functions)).To(Equal(8), "should return correct number of reconcile functions")
+	})
+
+	It("RequiredCrds function should return required crds", func() {
+		tt := getMockedTektonTasksOperand()
+		crds := tt.RequiredCrds()
+
+		Expect(len(crds) > 0).To(BeTrue(), "should return required crds")
+
+		for _, crd := range crds {
+			found := false
+			for _, c := range requiredCRDs {
+				if crd == c {
+					found = true
+				}
+			}
+			Expect(found).To(BeTrue(), "should return correct required crd")
+		}
+	})
+
```

Comment:
Can you add more unit tests that check reconciliation? Similarly to other operands.

Reply:
Can we have a ticket for it, to do it later? I'd prefer to have it in a new PR.

Reply:
Ok

Reply:
I've added similar tests to other operands:

1. `It("Should create tekton-tasks resources", func() { ... })`
2. `It("should remove tekton-tasks resources on cleanup", func() { ... })`

Done.